Storage/SQL/SQLiteBookmarksHelpers.swift

```diff
@@ -23,10 +23,16 @@ public func titleForSpecialGUID(guid: GUID) -> String? {
 }
 
 extension SQLiteBookmarks: ShareToDestination {
-    public func addToMobileBookmarks(url: NSURL, title: String, favicon: Favicon?) -> Success {
-        return self.insertBookmark(url, title: title, favicon: favicon,
-                                   intoFolder: BookmarkRoots.MobileFolderGUID,
-                                   withTitle: BookmarksFolderTitleMobile)
+    public func addToMobileBookmarks(url: NSURL, title: String, favicon: Favicon?) {
```

Comment:
You can — and should — still return `Success` here. Like this:

```
return self.isBookmarked(url.absoluteString, Direction.Local)
    >>== { yes in
        guard !yes else { return succeed() }
        return self.insertBookmark(…);
    }
```

Note that the chaining/flatMap operator, `>>==`, does the success check for you. A sequence like `a() >>== b >>== c` means "do these three things in order, passing values from one to the next, until failure or the last one succeeds, and return that failure or success".

`>>>` is the same but for success/failure only.